core/src/main/java/software/amazon/awssdk/auth/Aws4Signer.java

```diff
@@ -272,24 +270,19 @@ public SdkHttpFullRequest presign(Context.BeforeTransmission execution,
      * .amazon.com/general/latest/gr/sigv4-create-canonical-request.html to
      * generate the canonical request.
      */
-    private String createCanonicalRequest(SdkHttpRequest request,
+    private String createCanonicalRequest(SdkHttpFullRequest.Builder request,
```

Comment:
Don't know yet where this is being called but feels weird to pass in a mutable request object to signing rather than a fully built request.

Reply:
We pass in the immutable http request to the signer. The signer is implemented by calling toBuilder(), modifying it, then calling build(). This is one of the methods involved in the "modifying it" process.

Before this CR, the `SdkHttpRequest` was actually the builder as well. The builder implemented `SdkHttpRequest`. I changed that with this CR, so I changed the method signature